### ISSUE TITLE
distutils.spawn.find_executable not working to find ffmpeg

### ISSUE THREAD

**Issue by hubutui** [pytorch/vision#5837]
### 🐛 Describe the bug

distutils.spawn.find_executable not working to find ffmpeg

### Versions

https://github.com/pytorch/vision/blob/e8cb0bacd86c49e67a7e1a5f83c6da866bc451cf/setup.py#L343
for python 3.10,
```python
import distutils
distutils.spawn.find_executable("ffmpeg")
```
outputs:
```text
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    distutils.spawn.find_executable("ffmpeg")
AttributeError: module 'distutils' has no attribute 'spawn'
```
so `ffmpeg_exe = distutils.spawn.find_executable("ffmpeg")` leads to `ffmpeg_exe = None`
use this instead:
```python
from distutils import spawn
spawn.find_executable('ffmpeg')
```
also,
```text
DeprecationWarning: The distutils package is deprecated an
d slated for removal in Python 3.12. Use setuptools or check PEP 632 for potent
ial alternatives

```

**Comment by bjuncek** [comment]
@datumbox thanks for flagging, I'll take the liberty to replace this with `shutil.which` as it's recommended in [PEP632 migration advice](https://peps.python.org/pep-0632/#id11)

**Comment by bjuncek** [comment]
Closed via #5849
